### ISSUE TITLE
Update README

### ISSUE THREAD

**Issue by Lecrapouille** [bernedom/SI#112]
- Hello ! First line:

```
A header only c++ library that provides type safety
```

Could be nice to explicitly say it is C++17 directly on the 1st line (else said on section "building & compat" which is far away).

- The illustrative example does not compile (g++ --std=c++17 foo.cpp). I git cloned the HEAD:
```
foo.cpp:9:46: error: ‘const struct SI::detail::unit_t<'M', std::ratio<1>, long double, std::ratio<1> >’ has no member named ‘as’
    9 | constexpr auto thousand_grams = one_kilogram.as<SI::gram_t>();
      |                                              ^~
foo.cpp:9:59: error: missing template arguments before ‘>’ token
    9 | constexpr auto thousand_grams = one_kilogram.as<SI::gram_t>();
      |                                                           ^
foo.cpp:9:61: error: expected primary-expression before ‘)’ token
    9 | constexpr auto thousand_grams = one_kilogram.as<SI::gram_t>();
      |  
```

**Comment by bernedom** [comment]
You're right the examples does no longer compile since release 2.5.0 which removed some implicit conversions. 

At the moment the correct code would be 
`constexpr auto thousand_grams = one_kilogram.as<SI::gram_t<long double>>();`

which is not that nice. Will try to fix it

**Comment by bernedom** [comment]
The issue is fixed in this branch https://github.com/bernedom/SI/tree/feature/add-tests-for-as-conversion which will go in the next release.

**Comment by bernedom** [comment]
Released with version 2.5.1.
